lib/services/app-scaffolding-extensions-service.ts

```diff
@@ -41,6 +47,30 @@ export class AppScaffoldingExtensionsService extends dependencyExtensionsService
 		}).future<void>()();
 	}
 
+	public prepareDependencyExtension(dependencyExtensionName: string, dependencyConfig: IDependencyConfig, afterPrepareAction?: () => IFuture<void>): IFuture<void> {
+		return (() => {
+			let extensionVersion = this.getExtensionVersion(dependencyExtensionName);
+			let cachedVersion = extensionVersion || AppScaffoldingExtensionsService.DEFAULT_CACHED_VERSION;
+			let downloadUrl = this.$config.ON_PREM ? `${this.$config.AB_SERVER}/downloads/sb/generators/${dependencyExtensionName}/${dependencyConfig.version}` : `${AppScaffoldingExtensionsService.SCREEN_BUILDER_BUCKET_NAME}/v${dependencyConfig.version}/${dependencyExtensionName}.zip`;
+
+			this.$logger.trace("prepareDependencyExtension: Download url: %s, cached version", downloadUrl, cachedVersion);
```

Comment:
This code will actually also work because ot `util.format`'s behavior but I'll fix it for the sake of consistency - nice catch :fishing_pole_and_fish: